src/environments/array.js

```diff
@@ -130,6 +157,16 @@ const htmlBuilder = function(group, options) {
     const arstrutDepth = 0.3 * arrayskip;  // \@arstrutbox in lttab.dtx
 
     let totalHeight = 0;
+
+    // Set a position for \hline(s) at the top of the array, if any.
+    if (numHLinesBeforeRow[0] > 0) {
+        hlinePos.push(0);
+    }
+    for (let i = 2; i <= numHLinesBeforeRow[0]; i++) {
+        totalHeight += 0.25;
+        hlinePos.push(totalHeight);
+    }
```

Comment:
This can be simplified to (same for below):
```js
    if (numHLinesBeforeRow[0] > 0) {
        totalHeight += 0.25 * (numHLinesBeforeRow[0] - 1);
        hlinePos.push(totalHeight);
    }
```

Reply:
@ylemkimon We need a loop in order to handle multiple `\hline`s. So I think you mean that the code could read something like:
```
    for (let i = 1; i <= numHLinesBeforeRow[0]; i++) {
        if (i >  1) {
            totalHeight += 0.25;
        }
        hlinePos.push(totalHeight);
    }
```

That would be simpler for a human to read. But force of habit makes me try to avoid putting conditional statements inside a loop. Maybe this is not the place for that kind of thing. After all, we're not likely to get very many `\hline`s. 

Would you prefer my suggested revision to the current code?

Reply:
@ronkok I thought `hlinePos` is added only once, sorry. I think we can leave as-is.

Reply:
> I thought `hlinePos` is added only once

Which goes to show that code I suggested above would indeed be easier for a human to read. The speed loss with so few `\hline`s is negligible. I'll make the change.